### PR TITLE
Add tracing of recovery

### PR DESCRIPTION
**Pull request by GoPavel**

### CODE REVIEW

File: lib/merlin_recovery.ml [hunk 1]
```diff
@@ -226,6 +321,8 @@ struct
     (!shifted, final, candidates)
 
   let generate env =
+    Printer.print "Generate candidates for env:";
```
Comment (XJIE6) [inline lib/merlin_recovery.ml:324]: Better `"Generate candidates for env:\nStack:\n"`

File: lib/merlin_recovery.ml [hunk 2]
```diff
@@ -120,7 +168,48 @@ struct
...
+
+            let print_candidate x =
+              print @@ Printf.sprintf "Candidate #%d\n" !num; num := !num + 1;
+              print "Stack:\n";
+              print_env x.env;
+              print_current_state x.env;
+              print "\n"
+
+            let print_fail () =
+              print ">>>> Recovery failed"
```
Comment (XJIE6) [inline lib/merlin_recovery.ml:197]: Add some `\n`

File: lib/merlin_recovery.ml [hunk 3]
```diff
@@ -120,7 +168,48 @@ struct
...
+
+                (* print "Stack:"; *)
+                (* print_env r; *)
+                print "\n\n"
+
+            let print_candidate x =
+              print @@ Printf.sprintf "Candidate #%d\n" !num; num := !num + 1;
+              print "Stack:\n";
+              print_env x.env;
+              print_current_state x.env;
```
[the remark this answers is not in the document]
Comment (GoPavel) [reply lib/merlin_recovery.ml:193]: fixed

File: lib/merlin_recovery.ml (hunk 2 above)
Comment (GoPavel) [reply lib/merlin_recovery.ml:197]: fixed

**Pull request merged**